crates/vision/src/luminance_image_extractor.rs

```diff
@@ -0,0 +1,58 @@
+use std::num::NonZeroU32;
+
+use color_eyre::Result;
+use context_attribute::context;
+use fast_image_resize::{DynamicImageView, FilterType, ImageView, ResizeAlg, Resizer};
+use framework::MainOutput;
+use types::{grayscale_image::GrayscaleImage, nao_image::NaoImage};
+
+use crate::CyclerInstance;
+
+pub struct LuminanceImageExtractor {}
```

Comment:
Is extractor the right name?

Reply:
I don't know? Is it?

Reply:
I thought we are maybe transforming, converting, reinterpreting, ... Or is it really extracting? What does it extract? 🤔

Reply:
I briefly discussed with Gleske, we are not happy with either naming. Both not with the current naming and I am also not in favor of `LuminanceImageTransformer` as it sounds like a luminance image is going to be transformed, but it actually is the result of the producer. I would vote to keep it like it is, hoping that it is going to be absorbed by the robot detection

Reply:
Okay, fine for me. Then extractor seems to be a good name.